rust/src/lib/route.rs

```diff
@@ -179,6 +182,12 @@ pub struct RouteEntry {
     /// Serialize and deserialize to/from `route-type`.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub route_type: Option<RouteType>,
+    /// Congestion window clamp
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub cwnd: Option<u32>,
+    /// TCP metric lock flags. At the moment, they are for internal use only.
+    #[serde(skip)]
+    pub(crate) tcp_metric_lock_flags: Option<u32>,
```

Comment:
Even though I think this lock flags is kernel internal implementation on RCU. But since it require userspace to set it, I am OK to mark it as `pub(crate)` and expose or remove in the future.

Reply:
I actually would like to expose it in `nmstatectl  show` as `lock-cwnd: true` to be consistent with the behavior of NM and ip route. What do you think?


```
[root@fcvm rust]# nmcli c show eth1 | grep lock
ipv4.routes:                            { ip = 198.51.100.0/24, nh = 192.0.2.1, mt = 150 cwnd=20 lock-cwnd=true table=254 }
[root@fcvm rust]# ip -4 route
198.51.100.0/24 via 192.0.2.1 dev eth1 proto static metric 150 cwnd lock 20 
 ```

Reply:
I don't think it's a good idea. In order to set `cwnd` and not being useless, you need to set `lock-cwnd` too. I see it as a useless step that some users might not be aware of, so automatically setting `lock-cwnd` when the user sets `cwnd` in the yaml file seems a good idea to me.

Showing it when executing `nmstatectl show` doesn't sound bad, at first, but I think it might be confusing to show a field that you cannot set with `apply`.

Reply:
1. Nmstate schema is user facing or story driven. There is no point of hiding `tcp_metric_lock_flags` to user, but need backend to set it. The nispor and nm backend should set `cwnd` value property and set it as `None` if lock flag missing.
2. In kernel, this is a bit filed of u32, if we have `tcp_metric_lock_flags`, we might have 10+ properties for this single flags in the future.